service/storage_service.cc

```diff
@@ -3443,6 +3443,9 @@ future<> storage_service::decommission() {
     return run_with_api_lock(sstring("decommission"), [] (storage_service& ss) {
         return seastar::async([&ss] {
             ss.check_ability_to_perform_topology_operation("decommission");
+            if (ss._operation_mode != mode::NORMAL) {
+                throw std::runtime_error(::format("Node in {} state; wait for status to become normal or restart", ss._operation_mode));
+            }
```

Comment:
But we have similar check in the `raft_decommission()`:

```
        if (rs.state != node_state::normal) {
            throw std::runtime_error(::format("local node is not in the normal state (current state: {})", rs.state));
        }
```

Why doesn't it work?

Reply:
Probably because it requires a read barrier, which is not possible after the node leaves (it's banned)

Reply:
But there is a read barrier there, so it should hang or fail.